src/ui/citation-popup/index.ts

```diff
@@ -0,0 +1,42 @@
+import joplin from "api";
+import { DataStore } from "../../data/data-store";
+import constants from "../../constants";
+import { Reference } from "../../model/reference.model";
+const fs = joplin.require("fs-extra");
+
+let popupHandle: string = "";
+
+/**
+ * Show a dialog for the user to choose from a list of references
+ * to be inserted in the note content
+ */
+export async function showCitationPopup () {
+
+    // If the dialog was not initialized, create it and get its handle
+    if (popupHandle === "") {
+        popupHandle = await joplin.views.dialogs.create(constants.CITATION_POPUP_ID);
+    }
+
+    let html: string = await fs.readFile(
+        await joplin.plugins.installationDir() + "/ui/citation-popup/view.html",
+        'utf8'
+    );
+
+    const refs: Reference[] = DataStore.getAllReferences();
+    html = html.replace("<!-- content -->", fromRefsToHTML(refs));
+
+    await joplin.views.dialogs.setHtml(popupHandle, html);
+    await joplin.views.dialogs.open(popupHandle);
+}
+
+function fromRefsToHTML (refs: Reference[]): string {
+    const ans: string = (
+        `<ul>` +
+            refs
+                .map(ref => `<li>${ref.title}</li>`)
```

Comment:
You need to escape the title.

Reply:
Sorry, but I don't fully understand what that means. Do you mean that I need to print another attribute of the reference instead of the title ?

Reply:
If you add text into HTML, you need to escape it. Otherwise, for example, if the title contains an HTML tag, you'll end up with broken HTML. I usually use this package for this: https://www.npmjs.com/package/html-entities

Reply:
Ok, I understand now.